tests/frameworks/test_django.py

```diff
@@ -4,109 +4,114 @@
 import os
 
 import urllib3
+import pytest
+from typing import Generator
 from django.apps import apps
 from django.contrib.staticfiles.testing import StaticLiveServerTestCase
 
-from ..apps.app_django import INSTALLED_APPS
+from tests.apps.app_django import INSTALLED_APPS
 from instana.singletons import agent, tracer
-from ..helpers import fail_with_message_and_span_dump, get_first_span_by_filter, drop_log_spans_from_list
+from tests.helpers import fail_with_message_and_span_dump, get_first_span_by_filter, drop_log_spans_from_list
+from instana.instrumentation.django.middleware import url_pattern_route
 
 apps.populate(INSTALLED_APPS)
 
 
 class TestDjango(StaticLiveServerTestCase):
-    def setUp(self):
+    @pytest.fixture(autouse=True)
+    def _resource(self) -> Generator[None, None, None]:
         """ Clear all spans before a test run """
-        self.recorder = tracer.recorder
-        self.recorder.clear_spans()
         self.http = urllib3.PoolManager()
+        self.recorder = tracer.span_processor
+        self.recorder.clear_spans()
 
-    def tearDown(self):
+    def tearDown(self) -> None:
         """ Clear the INSTANA_DISABLE_W3C_TRACE_CORRELATION environment variable """
         os.environ["INSTANA_DISABLE_W3C_TRACE_CORRELATION"] = ""
 
-    def test_basic_request(self):
-        with tracer.start_active_span('test'):
+    def test_basic_request(self) -> None:
+        with tracer.start_as_current_span("test"):
             response = self.http.request('GET', self.live_server_url + '/', fields={"test": 1})
 
-        self.assertTrue(response)
-        self.assertEqual(200, response.status)
+        assert response
+        assert 200 == response.status
```

Comment:
It's not critical, but I suggest you always put the expected value on the right side of the statement. That way, if you want to change this value, it will be easier to modify, and it will also be easier to read the code with them in that part.
```suggestion
        assert response.status == 200
```

Reply:
Yeah I did that for flask tests when I manually changed all of them which was very time consuming so now I'm using vscode's find and replace and regex to change assert statements based on unittest to pytest which keeps the values in the same order as before.